src/extension.ts

```diff
@@ -452,3 +457,119 @@ async function restartLanguageServer(languageClient: LanguageClient = g_language
+        }
+    }
+
+    const hasJuliaup = await context.getActiveJuliaupExecutableAsync()
+    const showJuliaupInstallHint = vscode.workspace.getConfiguration('julia').get('juliaup.install.hint')
+    if (!hasJuliaup && showJuliaupInstallHint) {
+        await installJuliaOrJuliaup('juliaup')
+    }
+}
+
+async function installJuliaOrJuliaup(software: string = 'julia') {
+    // software can be 'julia' or 'juliaup'
+
+    const download = 'Download and Install'
+    const customCommand = 'Custom Command'
+    const options: string[] = [download, customCommand]
+
+    // Options for Julia
+    const configurePath = 'Configure path'
+    let message = 'It seems that julia is not installed, do you want to install it automatically using juliaup or you can manually add the path?'
+    let taskId = 'installJuliaTask'
+    let taskName = 'Install Julia'
+
+    options.push(configurePath)
+
+    // Options for Juliaup
+    const doNotShow = 'Do not show again'
+    if (software === 'juliaup') {
+        options.pop()
+
+        message = 'It seems that juliaup is not installed, do you want to install it automatically?'
+        taskId = 'installJuliaupTask'
+        taskName = 'Install Juliaup'
+
+        options.push(doNotShow)
+    }
+
+    const choice = await vscode.window.showInformationMessage(message, ...options)
+
+    if (choice === download) {
+        return await installJuliaOrJuliaupTask(taskId, taskName)
+    } else if (choice === customCommand) {
+        const command = await vscode.window.showInputBox({
+            placeHolder: 'Enter command',
+            validateInput: (value) => {
+                // return null if validates
+                return value.trim() !== '' ? null : 'Command is not valid!'
+            }
+        })
+
+        if (!command) {
+            // We return 1 as an exit code
+            return 1
+        }
+
+        return await installJuliaOrJuliaupTask(taskId, taskName, true, command)
+    } else if (choice === configurePath) {
+        vscode.commands.executeCommand('workbench.action.openSettings', 'julia.executablePath')
+    } else if (choice === doNotShow) {
+        vscode.workspace.getConfiguration('julia').update('juliaup.install.hint', false, vscode.ConfigurationTarget.Global)
+    }
+}
+
+async function installJuliaOrJuliaupTask(taskId: string, taskName: string, useCustomCommand: boolean = false, customCommand: string = '') {
+    let command = 'curl -fsSL https://install.julialang.org | sh -s -- -y'
+    if (process.platform === 'win32') {
+        command = 'winget install --name Julia --id 9NJNWW8PVKMN -e -s msstore'
+    }
+
+    if (useCustomCommand && customCommand === '') {
+        // We return 1 as an exit code
+        return 1
+    } else if (useCustomCommand) {
+        command = customCommand
+    }
+
+    const task = new vscode.Task(
+        {
+            type: 'julia',
+            id: taskId
+        },
+        vscode.TaskScope.Workspace,

```

Comment:
This will actually fail if you have no workspace open, which is very unfortunate and might break the whole `vscode.Task` approach... AFAICT, `vscode.TaskScope.Global` behaves the same...